subgraph/tests/keys.test.ts

```diff
@@ -194,6 +195,67 @@ describe('Change in expiration timestamp', () => {
 describe('Extend key', () => {
   test('should increase key timestamp', () => {
     mockDataSourceV11()
+
+    // mock and test
+    updateExpiration(BigInt.fromU64(expiration + 5000))
+    const newKeyExtended = createKeyExtendedEvent(
+      BigInt.fromU32(tokenId),
+      BigInt.fromU64(expiration + 5000),
+      false
+    )
+
+    handleKeyExtended(newKeyExtended)
+    assert.fieldEquals('Key', keyID, 'expiration', `${expiration + 5000}`)
+    dataSourceMock.resetValues()
+  })
+
+  test('should create receipt after key is extended on ERC20 lock', () => {
+    mockLockV11Erc20()
+    // create a key (that will create a receipt but that is not the one we want to test)
+    const newTransferEvent = createTransferEvent(
+      Address.fromString(nullAddress),
+      Address.fromString(keyOwnerAddress),
+      BigInt.fromU32(tokenId)
+    )
+    handleTransfer(newTransferEvent)
+
+    // mock and test
+    updateExpiration(BigInt.fromU64(expiration + 5000))
+
+    const newKeyExtended = createKeyExtendedEvent(
+      BigInt.fromU32(tokenId),
+      BigInt.fromU64(expiration + 5000),
+      true /** create ERC20 transfer event */
+    )
+
+    // check that receipt is created after key is extended
+    handleKeyExtended(newKeyExtended)
+
+    const hash = newKeyExtended.transaction.hash.toHexString()
+    const sender = newKeyExtended.transaction.from.toHexString()
+    const payer = newKeyExtended.transaction.from.toHexString()
+    const amount = newKeyExtended.transaction.value
+
+    // test values for not ERC20
+    assert.assertNotNull(newKeyExtended)
+    assert.entityCount('Receipt', 1)
+    assert.fieldEquals('Receipt', hash, 'id', hash)
+    assert.fieldEquals('Receipt', hash, 'lockAddress', lockAddress)
+    assert.fieldEquals(
+      'Receipt',
+      hash,
+      'timestamp',
+      `${newTransferEvent.block.timestamp}`
+    )
+    assert.fieldEquals('Receipt', hash, 'sender', sender)
+    assert.fieldEquals('Receipt', hash, 'payer', payer)
+    assert.fieldEquals('Receipt', hash, 'amountTransferred', `${amount}`)
```

Comment:
These are wrong. Here since this is an ERC20 lock, some ERC20 was transferred and the payer and amount should be based on that, not on the `newKeyExtended` values.